File: 10174.py
for i in range(int(input())):
    s = input().upper()

    correct = True
    if len(s) % 2:
        for j in range(len(s) // 2):
            if s[j] != s[len(s) - j - 1]:
                correct = False
    else:
        for j in range(len(s) // 2 + 1):
            if s[j] != s[len(s) - j - 1]:
                correct = False

    if correct: print("Yes")
    else: print("No")

File: 10174.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
	int n; cin >> n;
	getchar();

	for (int i = 0; i < n; i++) {
		string s; getline(cin, s);
		for (int j = 0; j < s.length(); j++)
			if (isalpha(s[j]))
				s[j] = tolower(s[j]);

		bool correct = true;
		if (s.length() % 2) {
			for (int j = 0; j < s.length() / 2; j++)
				if (s[j] != s[s.length() - j - 1])
					correct = false;
		}
		else {
			for (int j = 0; j <= s.length() / 2; j++)
				if (s[j] != s[s.length() - j - 1])
					correct = false;
		}

		if (correct) cout << "Yes\n";
		else cout << "No\n";
	}
}
